lectures/supply_demand_heterogeneity.md

```diff
@@ -156,98 +136,64 @@ c_{i}=\Pi^{-1}b_{i}-(\Pi^{\top}\Pi)^{-1}\mu_{i}p
 $$
 
 
+### Designing some Python code
 
-## Deducing a Representative Consumer
-
-In the class of multiple consumer economies that we are studying here, it turns out that there
-exists a single **representative consumer** whose preferences and endowments can be deduced from lists of preferences and endowments for the separate individual consumers.
-
-Consider a multiple consumer economy with initial distribution of wealth $W_i$ satisfying $\sum_i W_{i}=0$
-
-We allow an initial redistribution of wealth.
-
-We have the following objects
 
+Below we shall construct a Python class with the following attributes:
 
-- The demand curve:
-  
-$$ 
-c_{i}=\Pi^{-1}b_{i}-(\Pi^{\top}\Pi)^{-1}\mu_{i}p 
-$$
-
-- The marginal utility of wealth:
-  
-$$ 
-\mu_{i}=\frac{-W_{i}+p^{\top}\left(\Pi^{-1}b_{i}-e_{i}\right)}{p^{\top}(\Pi^{\top}\Pi)^{-1}p}
-$$
-
-- Market clearing:
-  
-$$ 
-\sum c_{i}=\sum e_{i}
-$$
-
-Denote aggregate consumption $\sum_i c_{i}=c$ and $\sum_i \mu_i = \mu$.
-
-Market clearing requires
-
-$$ 
-\Pi^{-1}\left(\sum_{i}b_{i}\right)-(\Pi^{\top}\Pi)^{-1}p\left(\sum_{i}\mu_{i}\right)=\sum_{i}e_{i}
-$$
-which, after a few steps, leads to
+ * **Preferences** in the form of
 
-$$
-p=\mu^{-1}\left(\Pi^{\top}b-\Pi^{\top}\Pi e\right)
-$$
+     * an $n \times n$  positive definite matrix $\Pi$
+     * an $n \times 1$ vector of bliss points $b$
 
-where
+ * **Endowments** in the form of
 
-$$ 
-\mu = \sum_i\mu_{i}=\frac{0 + p^{\top}\left(\Pi^{-1}b-e\right)}{p^{\top}(\Pi^{\top}\Pi)^{-1}p}.
-$$
+     * an $n \times 1$ vector $e$
+     * a scalar "wealth" $W$ with default value $0$
 
-Now consider the representative consumer economy specified above.
 
-Denote the marginal utility of wealth of the representative consumer by $\tilde{\mu}$.
+The class will include a test to make sure that $b \gg \Pi e $ and raise an exception if it is violated
+(at some threshold level we'd have to specify).
 
-The demand function is
+ * **A Person** in the form of a pair that consists of
 
-$$
-c=\Pi^{-1}b-(\Pi^{\top}\Pi)^{-1}\tilde{\mu} p
-$$
+    * **Preferences** and **Endowments**
 
-Substituting this into the budget constraint gives
+ * **A Pure Exchange Economy** will consist of
 
-$$
-\tilde{\mu}=\frac{p^{\top}\left(\Pi^{-1}b-e\right)}{p^{\top}(\Pi^{\top}\Pi)^{-1}p}
-$$
+    * a collection of $m$ **persons**
 
-In an equilibrium $c=e$, so
+       * $m=1$ for our single-agent economy
+       * $m=2$ for our illustrations of a pure exchange economy
 
-$$
-p=\tilde{\mu}^{-1}(\Pi^{\top}b-\Pi^{\top}\Pi e)
-$$
+    * an equilibrium price vector $p$ (normalized somehow)
+    * an equilibrium allocation $c_1, c_2, \ldots, c_m$ -- a collection of $m$ vectors of dimension $n \times 1$
```

Comment:
Chang from  $c^1, c^2, \ldots, c^m$ to $c_1, c_2, \ldots, c_m$ aligned with math notations above.

Reply:
Good catch @shlff